src/utils.ts

```diff
@@ -14,3 +14,7 @@ export function assertRequired<T>(object: T, ...properties: (keyof T & string)[]
     }
   })
 }
+
+export function checkRequired<T>(object: T, ...properties: (keyof T & string)[]) {
```

Comment:
we already have assertRequired

Reply:
assert required is not flexible enough. It immediately throws and don't support OR conditions. Check required enables us to only validate the conditions